### PR TITLE
Add a support of Apache Avro format

### PR DESCRIPTION
**Pull request by martinkubajda**

### CODE REVIEW

File: bspump/avro/source.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from avro.datafile import DatumReader
+import logging
+import avro
+import os
+from ..abc.source import TriggerSource
+
+# TODO: class AvroSource(TriggerSource)
```
Comment (ateska) [inline bspump/avro/source.py]: Well, obviously a missing Source ;-)

**Comment by ateska** [comment]
Hi there, approved (well, I see missing source).
Do you want me to merge it now?
@martinkubajda

**Comment by martinkubajda** [comment]
Hi, I implemented basic source and refactored sink but I want to discuss few things with you before merge, especially schemaless writer and reader

File: bspump/avro/sink.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		}
+	
+	File Types:
+	.avro - Avro Serialized Data
+	.avsc - Avro Schema
+		
+	"""
+	ConfigDefaults = {
+		'file_name_template': './sink{index}.avro',
+		'file_index_type': 'chunks',  # or 'datetime' or none
```
Comment (ateska) [inline bspump/avro/sink.py]: only `chunks` and `none` are implemented, please adjust the comment.

File: bspump/avro/sink.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+				L.warning('events_per_file % events_per_chunk needs to be zero. events_per_file was rounded')
+			elif self.EventsPerFile < self.ChunkSize:
+				self.ChunksPerFile = 1
+				L.warning('events_per_file needs to be equal or greater than events_per_chunk, events_per_file changed to events_per_chunk value')
+			else:
+				self.ChunksPerFile = int(self.Config['events_per_file'] / self.ChunkSize)
+			self.Chunks = 0
+			self.Index = 0
+
+		else:
```
Comment (ateska) [inline bspump/avro/sink.py]: Must be `elif ... == "none":` and then `else` with L.warning for unknown method.

**Pull request merged**